### PR TITLE
Akmal / feat: add useContractsForCompanyHook

### PR DESCRIPTION
**Pull request by akmal-deriv**
## Changes:

This feature introduces a new React hook, useContractsForCompanyHook, designed to fetch and manage contract data specific to a company.

- Data Fetching: The hook retrieves contract information for a given company, streamlining access to this data within components.
- State Management: Efficiently manages the state of the fetched contract data, ensuring it is readily available and up-to-date.
- Reusability: Provides a reusable solution for accessing company-specific contracts across different components in the project.
This hook simplifies and centralizes the process of handling company contract data, improving code organization and maintainability.

### CODE REVIEW

File: packages/trader/src/AppV2/Containers/Trade/trade-types.tsx [hunk 1]
```diff
@@ -178,14 +179,26 @@ const TradeTypes = ({ contract_type, onTradeTypeSelect, trade_types }: TTradeTyp
...
                 <Text size='sm' bold underlined>
                     {<Localize i18n_default_text='View all' />}
                 </Text>
             </a>
             <ActionSheet.Root isOpen={is_open} expandable={false} onClose={handleCloseTradeTypes}>
                 <ActionSheet.Portal>
-                    <ActionSheet.Header title={<Localize i18n_default_text='Trade types' />} />
+                    <ActionSheet.Header
+                        title={<Localize i18n_default_text='Trade types' />}
+                        icon={!is_editing && <Guide show_guide_for_selected_contract />}
```
Comment (kate-deriv) [inline packages/trader/src/AppV2/Containers/Trade/trade-types.tsx:200]: Flag show_guide_for_selected_contract  will show guide only for 1 trade type (currently selected). Do you want to achieve this or to show guide info for all trade types?

**Review by maryia-deriv — CHANGES_REQUESTED** [review]
some nit-picks, The rest LGTM :)

File: packages/trader/src/AppV2/Containers/Trade/trade-types.tsx (hunk 1 above)
Comment (akmal-deriv) [reply packages/trader/src/AppV2/Containers/Trade/trade-types.tsx:200]: I guess the other one

File: packages/trader/src/Stores/Modules/Trading/trade-store.ts [hunk 2]
```diff
@@ -1952,6 +1954,10 @@ export default class TradeStore extends BaseStore {
         this.has_symbols_for_v2 = !!active_symbols.length;
     }
 
+    setContractTypesListV2(contract_types_list: TContractTypesList) {
+        this.contract_types_list_V2 = contract_types_list;
```
[the remark this answers is not in the document]
Comment (akmal-deriv) [reply packages/trader/src/Stores/Modules/Trading/trade-store.ts:1958]: this doesn't matter)

[the remark this answers is not in the document]
Comment (maryia-deriv) [reply packages/trader/src/Stores/Modules/Trading/trade-store.ts:1958]: if you like a mix of camel & snake, sure mr. DJ :D

**Review by kate-deriv — APPROVED** [review]

**Review by maryia-deriv — APPROVED** [review]

File: packages/trader/src/Stores/Modules/Trading/trade-store.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (akmal-deriv) [reply packages/trader/src/Stores/Modules/Trading/trade-store.ts]: honestly, it doesn't matter to me what case it has, this doesn't impact functionality anyhow, and we shouldn't have `V2` in the future, so it's a temporary naming anyways, but if it's bothering you that much, I can change 😂

**Pull request merged**